### DISCUSSION TITLE
what kernel should be used for the bulk free energy part of Moelans (2011) paper

### DISCUSSION THREAD

**Discussion by yixishen** [idaholab/moose#22375] [Q&A Modules: Phase field] [answered]
Hi there, 

I am trying to apply the approach in Moelans(2011) paper Acta Material 59 (2011) 1077-1086 on multi-phase model. 

Attached is the evolution of the non-conserved order parameter. The first part is used to describe surface free energy and the second circled part is used to describe bulk free energy
<img width="741" alt="Screen Shot 2022-10-11 at 8 34 10 PM" src="https://user-images.githubusercontent.com/66687381/195222735-177c58d5-6a7f-4c6c-b486-e64852ee7656.png">
I have found kernel ACGrGrMulti can be used for surface free energy part, but I have not found any kernel for the bulk energy part.

Any idea for this?

Best

Yixi

**Comment by GiudGiud** [comment, marked as answer]
@dschwen @laagesen

**Reply by laagesen** [reply]
Hi Yixi, unfortunately we do not currently have any existing kernels for the boxed part of the equation. You would need to write some code to implement this. We would welcome a contribution if you do so!
